go/libraries/doltcore/schema/super_schema.go

```diff
@@ -0,0 +1,319 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/liquidata-inc/dolt/go/libraries/utils/set"
+)
+
```

Comment:
@zachmu I changed SuperSchema to track the last name used for a column. I added tests below in `schema/super_schema_test`

Reply:
SuperSchema is used to create schemas for the Diff and History tables. When we query the Diff and History tables, we want to see the most recently used column name